chainer/training/trainer.py

```diff
@@ -296,6 +298,16 @@ def run(self):
                     for name, entry in extensions:
                         if entry.trigger(self):
                             entry.extension(self)
+        except Exception as e:
+            if show_loop_exception_msg:
+                # Show the exception here, as it will appear as if chainer
+                # hanged in case any finalize method below deadlocks.
+                print("Exception in main training loop: {}".format(e))
+                print("Traceback (most recent call last):")
+                traceback.print_tb(sys.exc_info()[2])
```

Comment:
I found that only this line is shown in stderr. Specify `file=sys.stdout` to show it in stdout, or use print function to show all other error messages in stderr. I feel stderr is better. How do you think?